playbooks/common/openshift-cluster/initialize_facts.yml

```diff
@@ -6,12 +6,140 @@
+    include_role:
+      name: openshift_facts
+
+  # TODO: Should this role be refactored into health_checks??
+  - name: Run openshift_sanitize_inventory to set variables
+    include_role:
+      name: openshift_sanitize_inventory
+
+  - name: Detecting Operating System from ostree_booted
+    stat:
+      path: /run/ostree-booted
+    register: ostree_booted
+
+  # Locally setup containerized facts for now
+  - name: initialize_facts set fact l_is_atomic
+    set_fact:
+      l_is_atomic: "{{ ostree_booted.stat.exists }}"
+
+  - name: initialize_facts set fact for containerized and l_is_*_system_container
+    set_fact:
+      l_is_containerized: "{{ (l_is_atomic | bool) or (containerized | default(false) | bool) }}"
+      l_is_openvswitch_system_container: "{{ (openshift_use_openvswitch_system_container | default(openshift_use_system_containers | default(false)) | bool) }}"
+      l_is_node_system_container: "{{ (openshift_use_node_system_container | default(openshift_use_system_containers | default(false)) | bool) }}"
+      l_is_master_system_container: "{{ (openshift_use_master_system_container | default(openshift_use_system_containers | default(false)) | bool) }}"
+      l_is_etcd_system_container: "{{ (openshift_use_etcd_system_container | default(openshift_use_system_containers | default(false)) | bool) }}"
+
+  - name: initialize_facts set facts for l_any_system_container
+    set_fact:
+      l_any_system_container: "{{ l_is_etcd_system_container or l_is_openvswitch_system_container or l_is_node_system_container or l_is_master_system_container }}"
+
+  - name: initialize_facts set fact for l_etcd_runtime
+    set_fact:
+      l_etcd_runtime: "{{ 'runc' if l_is_etcd_system_container else 'docker' if l_is_containerized else 'host' }}"
+
+  # TODO: Should this be moved into health checks??
+  # Seems as though any check that happens with a corresponding fail should move into health_checks
+  - name: Validate python version - ans_dist is fedora and python is v3
+    fail:
+      msg: |
+        openshift-ansible requires Python 3 for {{ ansible_distribution }};
+        For information on enabling Python 3 with Ansible, see https://docs.ansible.com/ansible/python_3_support.html
+    when:
+    - ansible_distribution == 'Fedora'
+    - ansible_python['version']['major'] != 3
+
+  # TODO: Should this be moved into health checks??
+  # Seems as though any check that happens with a corresponding fail should move into health_checks
+  - name: Validate python version - ans_dist not Fedora and python must be v2
+    fail:
+      msg: "openshift-ansible requires Python 2 for {{ ansible_distribution }}"
+    when:
+    - ansible_distribution != 'Fedora'
+    - ansible_python['version']['major'] != 2
+
+  # TODO: Should this be moved into health checks??
+  # Seems as though any check that happens with a corresponding fail should move into health_checks
+  # Fail as early as possible if Atomic and old version of Docker
+  - block:
```

Comment:
Same as above.  Any `fail - msg` type checks are candidates for `health` or `preflight`.  I'd prefer to limit the number of dependency calls.  The attempt here is to do them upfront in the initial startup and then never do them again.  If they go into a role dependency then we have a tendency to include them in `mete/main.yml` for multiple roles.

We can definitely have the discussion about whether `health_checks` vs placing these types of checks into a `role` is superior.  Maybe a discussion for the architecture team.  I can see an argument for both sides but I'd prefer to have checks in one place so that in the future we know exactly where to place code like this.